tests/test_invenio_records_rest.py

```diff
@@ -81,6 +84,18 @@ def create_record(data):
     return pid, record
 
 
+def delete_record(pid, record):
+    with db.session.begin_nested():
+        record.delete()
```

Comment:
We should consider hooking PIDStore marking as deleted to record signal.

--
cc @lnielsen